src/components/App/LocationSearchModal/index.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react';
+import { useState, useEffect, useCallback } from 'react';
+import { Geocoder } from '@maptiler/geocoder';
+import styles from './LocationSearchModal.module.scss';
+import MapGeocoder from '../MapGeocoder';
+import type { MapAnchor, MapMarker } from '../../../types/map';
+import MapRenderer from '../MapRenderer';
+import { Button, Input } from '../../Global';
+import toast from 'react-hot-toast';
+
+const MAPTILER_ACCESS_TOKEN = process.env.NEXT_PUBLIC_MAPTILER_ACCESS_TOKEN;
+
+type LocationSearchModalProps = {
+  address: string;
+  latitude: string;
+  longtitude: string;
+  setAddress: React.Dispatch<React.SetStateAction<string>>;
+  setLatitude: React.Dispatch<React.SetStateAction<string>>;
+  setLongitude: React.Dispatch<React.SetStateAction<string>>;
+  setModalIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+const LocationSearchModal = ({
+  address,
+  latitude,
+  longtitude,
+  setAddress,
+  setLatitude,
+  setLongitude,
+  setModalIsOpen
+}: LocationSearchModalProps) => {
+  const [userMarker, setUserMarker] = useState<MapMarker>(null);
+  const [mapCenter, setMapCenter] = useState<MapAnchor>(null);
+  const [tempMarker, setTempMarker] = useState<MapMarker>(null);
+
+  useEffect((): void => {
+    navigator.geolocation.getCurrentPosition(
+      position => {
+        const anchor: MapAnchor = [
+          position.coords.latitude,
+          position.coords.longitude
+        ];
+        const zoom = 16;
+        setUserMarker({ anchor, zoom });
+        setMapCenter(anchor);
+      },
+      () => {
+        const anchor: MapAnchor = [49.88307, -119.48568];
```

Comment:
What are these coordinates? We should probably drop these in a constants file somewhere

Reply:
This is Matt's, I think it points to Kelowna

Reply:
Maybe move this array to a file and name it like `CENTRAL_KELOWNA` and use the constant in both cases @ketphan02

Reply:
Sure